Mathlib/Computability/TMComputable.lean

```diff
@@ -315,6 +315,21 @@ instance inhabitedTM2ComputableAux : Inhabited (TM2ComputableAux Bool Bool) :=
   ⟨(default : TM2Computable finEncodingBoolBool finEncodingBoolBool id).toTM2ComputableAux⟩
 #align turing.inhabited_tm2_computable_aux Turing.inhabitedTM2ComputableAux
 
+/-- Generalize to functions computable in polynomial time -/
+def PolyTimeFunction {α β: Type} {ea : FinEncoding α} {eb : FinEncoding β} (f : α → β) :=
+  @TM2ComputableInPolyTime α β ea eb f
+
+/-- Indicator function for a language decidable in polynomial time -/
+def PolyTimeLanguageIndicator {α: Type} {ea : FinEncoding α} {ebool : FinEncoding Bool}
+    (f : α → Bool) := @TM2ComputableInPolyTime α Bool ea ebool f
+
+/-- Relationship indicates that "x" has a certificate with encoded length polynomially bounded as
+a function of the encoded length of x. -/
+def verification {α γ: Type} {ea : FinEncoding α} {eg : FinEncoding γ} {p: Polynomial ℕ}
+    (certified : Rel α γ) : Prop :=
+  ∀ (x : α) (certificate : γ), certified x certificate →
+  (ea.encode x).length > p.eval (eg.encode certificate).length
```

Comment:
Why is this not `eg.encode certificate < p.eval (ea.encode x).length`

Reply:
Please see the correction--it is the certificate length that must be bounded as a function of the length of x. Thanks for pointing that out.